### DISCUSSION TITLE
Why my rsa key is still valid

### DISCUSSION THREAD

**Discussion by JeenHwahn** [community/community#40681] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

I know that the git hub does not support the rsa key, but the rsa ssh key I added last year can still be used. Now I can't push when I generate a pair of keys in the same way. Why?

**Comment by DSDmark** [comment]
Hi @Jeffrey-L1,

Did you using same system when you were generated that keys first time?
If you're using the same system, then you need to delete that keys first.

**RSA** isn't key is a cryptographic system which is used that **algorithm** and generate two **key** bases on the two large prime numbers. By this algorithm, there are two keys generate **public-key** which we called **encryption key** and the decryption key which is **private-key**.

**Reply by JeenHwahn** [reply]
Hi DSDmark,
    Thks,The secret key pair of my own computer has been used for several years. The pub file still starts with ssh-rsa. Today, I taught my friend how to use git on his computer. I found that the keys generated by the previous method are useless. I tried it on the virtual machine, but it didn't work. I'm curious that my native computer can only use the previous key pair without changing the encryption algorithm such as ed25519

**Reply by DSDmark** [reply]
@Jeffrey-L1, in my opinion, you can't use the same key pair. It's unique for each system. But, if you have that account access. Then, you could generate a new **ssh** key and add in the setting. By this way, you can use many systems by different ssh key.

![Setting of ssh](https://user-images.githubusercontent.com/85744816/205436725-3b669061-235a-4f67-a4d1-3fc76e455c91.png)

**Comment by airtower-luna** [comment]
> I know that the git hub does not support the rsa key, but the rsa ssh key I added last year can still be used.

That is incorrect, RSA keys are supported. What changed somewhat recently is that you can't add keys that are meant to be used with SHA-1 signatures, you have to use SHA-2. Support for **DSA** keys was removed at the same time. 

See: https://github.blog/changelog/2022-03-15-removed-unencrypted-git-protocol-and-certain-ssh-keys/

**Comment by JeenHwahn** [comment, marked as answer]
dear all，
    The putty key. ppk file generated by tortoisegit 2.11 just doesn't work.It will update to tortoisegit 2.13
